packages/devtools_app/lib/src/flutter/banner_messages.dart

```diff
@@ -0,0 +1,304 @@
+    with AutoDisposeMixin {
+  BannerMessagesController controller;
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    final newController = Controllers.of(context)?.bannerMessages;
+    if (newController == controller) return;
+    controller = newController;
+
+    autoDispose(controller.onRefreshMessages.listen((_) => setState(() {})));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final messagesForScreen = widget.screen.messages(context);
+    if (messagesForScreen.isNotEmpty) {
+      final messagesToShow = <Widget>[];
+      for (Widget message in messagesForScreen) {
+        assert(message is UniqueMessage);
+        if (!controller.isMessageDismissed((message as UniqueMessage).id)) {
+          messagesToShow.add(message);
+        }
+      }
+      if (messagesToShow.isNotEmpty) {
+        return Column(
+          children: messagesToShow,
+        );
+      }
+    }
+    return const SizedBox();
+  }
+}
+
+class BannerMessage extends StatelessWidget implements UniqueMessage {
+  const BannerMessage({
+    @required this.messageId,
+    @required this.textSpans,
+    @required this.backgroundColor,
+    @required this.foregroundColor,
+  });
+
+  final String messageId;
+  final List<TextSpan> textSpans;
+  final Color backgroundColor;
+  final Color foregroundColor;
+
+  @override
+  String get id => messageId;
+
+  @override
+  Widget build(BuildContext context) {
+    final bannerMessagesController = Controllers.of(context).bannerMessages;
+    return Card(
+      color: backgroundColor,
+      child: Padding(
+        padding: const EdgeInsets.all(defaultSpacing),
+        child: Row(
+          children: <Widget>[
+            Expanded(
+              child: Container(
+                child: RichText(
+                  text: TextSpan(
+                    children: textSpans,
+                  ),
+                ),
+              ),
+            ),
+            const SizedBox(width: denseSpacing),
+            CircularIconButton(
+              icon: Icons.close,
+              backgroundColor: backgroundColor,
+              foregroundColor: foregroundColor,
+              // TODO(kenz): animate the removal of this message.
+              onPressed: () => bannerMessagesController.dismissMessage(id),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _BannerError extends BannerMessage {
+  const _BannerError({
+    @required String id,
+    @required List<TextSpan> textSpans,
+    @required DevToolsScreenType screenType,
+  }) : super(
+          messageId: id,
+          textSpans: textSpans,
+          backgroundColor: devtoolsError,
+          foregroundColor: foreground,
+        );
+
+  static const foreground = Colors.white;
+  static const linkColor = Color(0xFF88D7FC);
+}
+
+// TODO(kenz): add "Do not show this again" option to warnings.
+class _BannerWarning extends BannerMessage {
+  const _BannerWarning({
```

Comment:
the warning banner looks good on a light theme but is too loud on a dark theme. Can we use the theme to get the foreground and background colors for this case or use a themed color with light and dark options.